sympy/physics/mechanics/body.py

```diff
@@ -101,6 +101,8 @@ def __init__(self, name, masscenter=None, mass=None, frame=None,
 
         self.name = name
         self._loads = []
+        self.is_rigidbody = None
+        self.is_particle = None
```

Comment:
This would need unit tests for the new attributes, as well as documentation and examples.

Why is this needed?

Reply:
Because `Body` inherits both `RigidBody` and `Particle` so Kane's method couldn't differentiate if `Body` is `RigidBody` and `Particle`.

Reply:
I have added tests.

Reply:
KanesMethod is not designed to work with Body. Either you pass the correct objects to KanesMethod or you modify KanesMethod to work with new objects. This seems like a hack not an actual fix. Duck typing with a try/except is all you'd need or even an isinstance check. Adding these attributes is superfluous.